services/venice-server/src/main/java/com/linkedin/venice/listener/ServerReadMetadataRepository.java

```diff
@@ -186,4 +232,76 @@ public ServerCurrentVersionResponse getCurrentVersionResponse(String storeName)
     }
     return response;
   }
+
+  private void checkStore(String storeName, Store store) throws VeniceException, UnsupportedOperationException {
+
+    // Check fast client compatibility
+    if (!store.isStorageNodeReadQuotaEnabled()) {
```

Comment:
I see why you added enabling of SN read quota in that get store properties test now. This check or gatekeeping logic should only be applicable to FC. i.e. if it's DVC or other clients like CC fetching metadata then we shouldn't check for SN read quota and throw exceptions if it's not enabled.

Reply:
Moving to the. "getMetadata" method